src/main/java/net/sf/jabref/external/push/AbstractPushToApplication.java

```diff
@@ -0,0 +1,196 @@
+/*  Copyright (C) 2015 JabRef contributors.
+    Copyright (C) 2015 Oscar Gustafsson.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.external.push;
+
+import java.io.IOException;
+
+import javax.swing.*;
+
+import net.sf.jabref.*;
+
+import com.jgoodies.forms.builder.FormBuilder;
+import com.jgoodies.forms.layout.FormLayout;
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.gui.actions.BrowseAction;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.database.BibtexDatabase;
+import net.sf.jabref.model.entry.BibtexEntry;
+
+/**
+ * Abstract class for pushing entries into different editors.
+ */
+public abstract class AbstractPushToApplication implements PushToApplication {
+
+    protected boolean couldNotCall; // Set to true in case the command could not be executed, e.g., if the file is not found
+    protected boolean couldNotConnect; // Set to true in case the tunnel to the program (if one is used) does not operate
+    protected boolean notDefined; // Set to true if the corresponding path is not defined in the preferences
+    protected JPanel settings;
+    protected final JTextField Path = new JTextField(30);
+    protected String commandPath;
+    protected String commandPathPreferenceKey;
+    protected String citeCommand = Globals.prefs.get(JabRefPreferences.CITE_COMMAND);
+    protected FormBuilder builder;
+
+
+    @Override
+    public String getName() {
+        return Localization.menuTitle("Insert selected citations into %d", getApplicationName());
+    }
+
+    @Override
+    public String getTooltip() {
+        return Localization.lang("Push to %0", getApplicationName());
+    }
+
+    @Override
+    public String getKeyStrokeName() {
+        return Localization.lang("Push to %0", getApplicationName());
```

Comment:
The key stroke name should not be localised, as this is used to get the shortcut key binding for this action.